daemon/containerd/image_snapshot_unix.go

```diff
@@ -85,7 +95,50 @@ func (i *ImageService) unremapRootFS(ctx context.Context, mounts []mount.Mount)
 				return err
 			}
 
-			return os.Lchown(path, uid, gid)
+			return chownWithCaps(path, uid, gid)
 		})
 	})
 }
+
+// chownWithCaps will chown path and preserve the extended attributes.
+func chownWithCaps(path string, uid int, gid int) error {
+	xattrKeys, err := sysx.LListxattr(path)
+	if err != nil {
+		if !errors.Is(err, syscall.EOPNOTSUPP) {
+			return err
+		}
+	}
+
+	xattrs := make(map[string][]byte, len(xattrKeys))
+
+	for _, xattr := range xattrKeys {
+		data, err := sysx.LGetxattr(path, xattr)
+		if err != nil {
+			if !errors.Is(err, syscall.EOPNOTSUPP) {
+				return err
+			}
+			break
+		}
+		xattrs[xattr] = data
+	}
+
+	if err := os.Lchown(path, uid, gid); err != nil {
+		return err
+	}
+
+	for xattrKey, xattrValue := range xattrs {
+		length := len(xattrValue)
+		if xattrValue[versionOffset] == vfsCapRevision3 {
+			xattrValue[versionOffset] = vfsCapRevision2
+			length = xattrCapsSz2
+		}
+		if err := sysx.LSetxattr(path, xattrKey, xattrValue[:length], 0); err != nil {
+			if !errors.Is(err, syscall.EOPNOTSUPP) {
+				return err
+			}
```

Comment:
I guess _technically_ the error return should be here (not when _reading_ the attributes), but in practice they should be symmetrical (you cannot _either_ read _and_ write them), but if we decide to produce an error, we should do for both. (similar to https://github.com/tonistiigi/fsutil/blob/797bd683d5a28a87bea5421f51db1e112ac2edd4/copy/copy_nowindows.go#L15C1-L32)